onnxruntime/wasm/api.cc

```diff
@@ -9,55 +9,192 @@
 #include <vector>
 
 namespace {
-Ort::Env* g_env;
+OrtEnv* g_env;
 }  // namespace
 
-void OrtInit(int numThreads, int logging_level) {
+OrtErrorCode CheckStatus(OrtStatusPtr status) {
+  OrtErrorCode error_code = ORT_OK;
+  if (status) {
+    std::string error_message = Ort::GetApi().GetErrorMessage(status);
+    error_code = Ort::GetApi().GetErrorCode(status);
+    std::cerr << Ort::Exception(std::move(error_message), error_code).what() << std::endl;
+    Ort::GetApi().ReleaseStatus(status);
+  }
+  return error_code;
+}
+
+int OrtInit(int numThreads, int logging_level) {
+  OrtLoggingLevel level = ORT_LOGGING_LEVEL_WARNING;
+  switch (logging_level) {
+    case 0:
+      level = ORT_LOGGING_LEVEL_VERBOSE;
+      break;
+    case 1:
+      level = ORT_LOGGING_LEVEL_INFO;
+      break;
+    case 2:
+      level = ORT_LOGGING_LEVEL_WARNING;
+      break;
+    case 3:
+      level = ORT_LOGGING_LEVEL_ERROR;
+      break;
+    case 4:
+      level = ORT_LOGGING_LEVEL_FATAL;
+      break;
+  }
+
 #if defined(__EMSCRIPTEN_PTHREADS__)
-  OrtThreadingOptions* tp_options;
-  Ort::ThrowOnError(Ort::GetApi().CreateThreadingOptions(&tp_options));
-  Ort::ThrowOnError(Ort::GetApi().SetGlobalIntraOpNumThreads(tp_options, numThreads));
-  Ort::ThrowOnError(Ort::GetApi().SetGlobalInterOpNumThreads(tp_options, 1));
+  int error_code = ORT_OK;
+
+  OrtThreadingOptions* tp_options = nullptr;
+  error_code = CheckStatus(Ort::GetApi().CreateThreadingOptions(&tp_options));
```

Comment:
how about create a new macro like `ReturnOnFailure()` for this

Reply:
not a fan of macro, but there are lots of redundancy due to C API error handling. let me introduce some macros here.

Reply:
not necessarily a macro. I think a function should work as well.

Reply:
need a return, so function is not working.